components/background_hang_monitor/background_hang_monitor.rs

```diff
@@ -20,18 +21,46 @@ pub struct HangMonitorRegister {
     sender: Sender<(MonitoredComponentId, MonitoredComponentMsg)>,
 }
 
+#[derive(Copy, Clone, PartialEq)]
+enum SamplerState {
+    NotSampling,
+    StartSampling(Duration),
+    Sampling,
+    Resolving,
+}
+
+lazy_static! {
+    static ref SAMPLING_STATE: Mutex<SamplerState> = Mutex::new(SamplerState::NotSampling);
+}
 impl HangMonitorRegister {
     /// Start a new hang monitor worker, and return a handle to register components for monitoring.
-    pub fn init(constellation_chan: IpcSender<HangAlert>) -> Box<BackgroundHangMonitorRegister> {
+    pub fn init(
+        constellation_chan: IpcSender<HangMonitorAlert>,
+    ) -> Box<BackgroundHangMonitorRegister> {
         let (sender, port) = unbounded();
         let _ = thread::Builder::new().spawn(move || {
-            let mut monitor = { BackgroundHangMonitorWorker::new(constellation_chan, port) };
+            let mut monitor = BackgroundHangMonitorWorker::new(constellation_chan, port);
             while monitor.run() {
                 // Monitoring until all senders have been dropped...
             }
         });
         Box::new(HangMonitorRegister { sender })
     }
+
+    pub fn toggle(rate: Duration) {
```

Comment:
I think there are two issues with this setup:

1. I think it's not going to work in multiprocess mode(since `toggle` is called from the "main" process where the embedder is running). 
2. When you call `toggle` one cannot be sure that the bhg is actually running, so `toggle` could return successfully but actually not result in anything changing. 
3. Nit: in practice there will only ever be one bhg per process, so the mutex around `SamplerState` is a pity.